packages/kbn-unified-data-table/src/components/data_table_additional_display_settings.tsx

```diff
@@ -124,14 +124,25 @@ export const UnifiedDataTableAdditionalDisplaySettings: React.FC<
   }
 
   if (onChangeSampleSize) {
+    let step = minRangeSampleSize === RANGE_MIN_SAMPLE_SIZE ? RANGE_STEP_SAMPLE_SIZE : 1;
+
+    if (
+      step > 1 &&
+      ((activeSampleSize && !checkIfValueIsMultipleOfStep(activeSampleSize, step)) ||
+        !checkIfValueIsMultipleOfStep(minRangeSampleSize, step) ||
+        !checkIfValueIsMultipleOfStep(maxAllowedSampleSize, step))
```

Comment:
One Question: I do not understand why do we need to check  that `minRangeSampleSize` and `maxAllowedSampleSize` are the multiples of `step`. Isn't checking just `activeSampleSize` enough?

```js
!checkIfValueIsMultipleOfStep(minRangeSampleSize, step) ||
        !checkIfValueIsMultipleOfStep(maxAllowedSampleSize, step))

```

Reply:
@logeekal No, it was not enough. The exception was actually coming from `maxAllowedSampleSize` being `9995`.

Reply:
Okay so just to understand, EUI will raised an error even if `activeSampleSize` which is currently selected by the user is a valid multiple of `step` but `maxAllowedSampleSize` is not? 

It is weird if it does that, right?

I thought the error was coming when we change `discover: sampleSize` option to `9995`, it means the `activeSampleSize` is also `9995`. May be i will need to look more closely.

Reply:
@logeekal If user changes `discover: sampleSize` to `9995` then both `value` and `max` will get `9995` in Discover popover. This could cause 2 crashes (if step is `10`):

1. On opening the popover because the `activeSampleSize` is not a valid multiple - this is where I started fixing the bug
2. I also thought initially that adding a check for the value (`activeSampleSize && !checkIfValueIsMultipleOfStep(activeSampleSize, step)`) should fix the problem but then I started seeing another similar crash. This time when user tries to change the value via UI:
![Sep-06-2024 14-26-39](https://github.com/user-attachments/assets/69736c16-727d-4789-b3e0-645ffd121545)

So Eui validates both: the value and [the max too](https://github.com/elastic/eui/blob/ba42930b0a0ae601389c3615be20f6782e0dbaef/packages/eui/src/components/form/range/range_track.tsx#L62). Well, on our side I ended up adding 3 checks for the value, max and min just to be sure 😀

Reply:
Thank you so much for your explanation and patience. 😀

This makes sense.